code/datums/gods/patrons/patrons_blessings.dm

```diff
@@ -0,0 +1,396 @@
+			if(!mending_amount)
+				return FALSE
+
+			if(!M || !M.reagents)
+				to_chat(usr, span_warning("[M] has no reagent container."))
+				return FALSE
+
+			M.reagents.add_reagent(/datum/reagent/medicine/stronghealth, mending_amount)
+			var/patron_name = (M.patron && istype(M.patron)) ? M.patron.name : "a divine presence"
+			to_chat(M, span_nicegreen("You feel your body renewing. [patron_name] has renewed your body with temporary divinity."))
+			to_chat(usr, span_notice("You mended [M] with [mending_amount] units of Lifeblood."))
+			log_admin("[key_name(usr)] mended [key_name(M)] with [mending_amount] units of Lifeblood.")
+			return TRUE
+
+	if(!blessing_path)
+		return FALSE
+
+	var/duration_choice = input("Select Duration for the Blessing:") as null|anything in list( \
+		"1 Minute", "5 Minutes", "10 Minutes", "20 Minutes", \
+		"30 Minutes", "60 Minutes", "Until Sleep", "Infinite")
+	if(!duration_choice)
+		return FALSE
+
+	var/until_sleep = FALSE
+	var/duration = -1
+	switch(duration_choice)
+		if("1 Minute") duration = 1 MINUTES
+		if("5 Minutes") duration = 5 MINUTES
+		if("10 Minutes") duration = 10 MINUTES
+		if("20 Minutes") duration = 20 MINUTES
+		if("30 Minutes") duration = 30 MINUTES
+		if("60 Minutes") duration = 60 MINUTES
+		if("Until Sleep") until_sleep = TRUE
+		if("Infinite") duration = -1
+
+	/// Apply Blessing
+	if(M.apply_status_effect(blessing_path))
+		message_admins(span_notice("Admin [key_name_admin(usr)] blessed [key_name_admin(M)] with [blessing_path]! Duration: [duration_choice]."))
+		log_admin("[key_name(usr)] blessed [key_name(M)] with [blessing_path] for [duration_choice].")
+
+		M.playsound_local(get_turf(M), 'sound/magic/bless.ogg', 100, FALSE)
+
+		var/flavor_text = get_patron_blessing_text(M, blessing_path)
+		if(flavor_text)
+			to_chat(M, span_nicegreen("[flavor_text]"))
+
+		/// Set actual expiration if needed (manual override for duration_modification types)
+		if(duration > 0)
+			var/datum/status_effect/active = M.get_status_effect(blessing_path)
+			if(active && istype(active, /datum/status_effect/buff/duration_modification))
+				active.duration = world.time + duration
+			M.start_blessing_duration_timer(blessing_path, duration)
+
+		if(until_sleep)
+			M.start_blessing_sleep_monitor(blessing_path)
+
+		var/alert_desc = flavor_text ? span_nicegreen("[flavor_text]") : span_nicegreen("A divine force blesses you!")
+		M.modify_blessing_alert_desc(blessing_path, alert_desc)
+
+		return TRUE
+
+	/// Toggle Off: Remove if present
+	else if(M.remove_status_effect(blessing_path))
+		message_admins(span_notice("Admin [key_name_admin(usr)] lifted blessing [blessing_path] from [key_name_admin(M)]!"))
+		log_admin("[key_name(usr)] lifted blessing [key_name(M)] from [blessing_path].")
+		return TRUE
+
+	return FALSE
+
+
+/// Starts a timer to auto-remove the blessing after duration expires
+/mob/living/proc/start_blessing_duration_timer(var/blessing_path, var/duration)
```

Comment:
it needs to remove the var/ defs inside the proc, it will work the same way but it is good practicesto not use it and the run liters is pointing that too.